src/libmusig2.h

```diff
@@ -7,212 +7,141 @@
- *                          : sxo_PK_LIST, the list of serialised public keys.
- *                          : L, concatenation of the serialised public keys.
- *                          : N, number of signers.
- *              : IN/OUT    : a_LIST, the ordered list of signers' exponents.
- * Returns      : 1/0
+typedef struct{
+    secp256k1_context* ctx;
+    secp256k1_pubkey X_;
+    secp256k1_pubkey R_;
+    unsigned char *L;
+    int state;
+}musig2_context;
+
+/** Struct      : musig2_context_sig
+ *  Purpose     : Stores the parameters of musig2 signer.
+ *  Parameters  : mc: a musig2_context object including the parameters of musig2.
+ *              : commlist: Batch commitment list of a signer.
+ *              : keypair: Public and secret keys of signers.
+ *              : agg_R_list: The list of aggregated batch commitments.
  * */
-int MuSig2_KeyAggCoef(const secp256k1_context *ctx,
-                      unsigned char **ser_xonly_X_LIST, unsigned char **a_LIST, unsigned char *L,
-                      int N ) ;
-/*----------------------------------------------------------------------------------------------------------*/
-
-
-/** Function    : MuSig2_KeyAgg
- *  Purpose     : Computes the aggregated public key X_ for given list of public keys and corresponding exponent list.
- *                Each public key is multiplied with its exponent and the results are combined in X_. The x_only version
- *                of X_ is stored in xonly_X_ and serialised x_only version is stored in ser_xonly_X_. If x_only version is the negative
- *                of the aggregated public key, parity_X_ is set to 1.  If the public keys are aggregated successfully
- *                returns 1, 0 otherwise.
- *  Parameters  : IN        : ctx, a secp256k1_context object.
- *                          : X_LIST, the list of public keys.
- *                          : a_LIST, the list of exponents.
- *                          : N, number of signers.
- *              : IN/OUT    : X_, the aggregated public key.
- *                          : xonly_X_, the x_only aggregated public key.
- *                          : ser_xonly_X_, the serialized x_only aggregated public key.
- *                          : parity_X_, parity of xo_X_.
- * Returns      : 1/0
+typedef struct {
+    musig2_context *mc;
+    secp256k1_keypair **commlist;
+    secp256k1_keypair keypair;
+    secp256k1_pubkey agg_R_list[V];
+}musig2_context_sig;
+
+/** Struct      : musig2_param
+ *  Purpose     : Stores the parameters to generate a partial signature.
+ *  Parameters  : a: The exponent `a` of signer.
+ *              : b: nonce, b = H_non(X, (R_1, ..., R_V), msg).
+ *              : c: challenge, c = H_sig(X, R, msg).
+ *              : b_list: The list of `b` values, b_LIST = { b^(j-1) }.
+ *              : msg: The message to be signed.
+ *              : sx_X: Serialized x_only aggregated public key.
+ *              : sx_R: Serialized x_only aggregated R.
```

Comment:
we don't need to specify that the serialised representations are x_only. It suffices to say that it is a `serialised aggregated public key`. This holds everywhere in the code. I would remove the mention of x_only, as it feels like there is other way to serialise points different to `x_only` notation.